internal/drivers/filesystem/storage.go

```diff
@@ -0,0 +1,267 @@
+/******************************************************************************
+*
+*  Copyright 2022 ruilopes.com
+*
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+******************************************************************************/
+
+// TODO the error returned here are displayed to the user as-is. is that a problem? like, leaking path names?
+// TODO wrap all the errors?
```

Comment:
I'm mostly fine with leaving errors as-is, but I agree that path names should have the `rootPath` obscured. If you take my other suggestion of using the stdlib functions for filesystem access, that should be easy to add since all those functions always return `*os.PathError` which has the Path in a separate field for easy access and manipulation.

Reply:
I ended up leaving this as-is.